web/elm/src/Effects.elm

```diff
@@ -120,19 +146,22 @@ type Effect
     | SaveToken String
     | LoadToken
     | ForceFocus String
+    | OpenBuildEventStream { url : String, eventTypes : List String }
+    | CloseBuildEventStream
+    | CheckIsVisible String
 
 
 type ScrollDirection
-    = ToWindowTop
+    = ToTop
     | Down
     | Up
-    | ToWindowBottom
-    | Builds Float
-    | ToCurrentBuild
+    | ToBottom
+    | Element String Float
+    | ToBuild Int
```

Comment:
`ToBuild` now seems a bit overspecific, although I can see the argument for not moving it yet.